src/services/authenticator.js

```diff
@@ -71,6 +74,77 @@ function Authenticator() {
       });
     });
   };
+
+  this.loginWithEmailOrTokenArgv = async (config) => {
+    const token = await this.login(config);
+    this.saveToken(token);
+    return null;
+  };
+
+  this.loginWithToken = (token) => {
+    const validationResult = this.validateToken(token);
+    if (validationResult !== true) {
+      throw new Error(validationResult);
+    }
+    return token;
+  };
+
+  this.login = async ({ email, password, token }) => {
+    if (!email) email = await this.promptEmail();
+
+    if (token) {
+      return this.loginWithToken(token);
+    }
+
+    const isGoogleAccount = await api.isGoogleAccount(email);
+    if (isGoogleAccount) {
+      return this.loginWithGoogle();
+    }
+
+    return this.loginWithPassword(email, password);
+  };
+
+  this.promptEmail = async () => {
+    const { email } = await inquirer.prompt([{
+      type: 'input',
+      name: 'email',
+      message: 'What is your email address?',
+      validate: (input) => {
+        if (!Joi.string().email().validate(input).error) {
+          return true;
+        }
```

Comment:
We only validate email when prompted, not when sent via command line. We could consider validate it in all cases:

<img width="537" alt="Capture d’écran 2020-03-12 à 14 35 38" src="https://user-images.githubusercontent.com/1575946/76528523-3adf0d80-6471-11ea-832c-12b464a96cd2.png">

Reply:
Let me fix this